hazelcast/src/main/java/com/hazelcast/mapreduce/impl/operation/KeysAssignmentResult.java

```diff
@@ -57,27 +58,14 @@ public ResultState getResultState() {
     @Override
     public void writeData(ObjectDataOutput out)
             throws IOException {
-        out.writeBoolean(assignment != null);
-        if (assignment != null) {
-            out.writeInt(assignment.size());
-            for (Map.Entry<Object, Address> entry : assignment.entrySet()) {
-                out.writeObject(entry.getKey());
-                out.writeObject(entry.getValue());
-            }
-        }
+        writeNullableMap(assignment, out);
```

Comment:
The logic of the boolean null flag is inverted comparing to the original.

Reply:
good catch, inverted the flag in `SerializationUtil`